### PR TITLE
Add Night Time Calculation to Flight Conditions

### PR DESCRIPTION
**Pull request by jbethanco**
Add ability to get latitude and longitude from ICAO
-Add decodable AirfieldsStruct
-Add airfields json file, with intentionally wrong file extension (large json files crash Xcode if opened)

Add CLLocationCoordinate2D extension file with multiple calculating functions
Add extension on Double to go to and from radians and degrees
Add night calculator enum to hold the night calculating functions
Add button to auditor view to calculate the night hours
Add info button, and popover, to night hours calc button, to describe what it does

Simulator Screen Recording - _iPad Pro 9 7 v 14 5 - 2021-04-27 at 07 19 56

### CODE REVIEW

**Comment by cbrech** [comment]
Can we put the calculate button above like gonkulate is?

**Comment by jbethanco** [comment]
> Can we put the calculate button above like gonkulate is?

sure.

**Comment by jbethanco** [comment]
> > Can we put the calculate button above like gonkulate is?
> 
> sure.

done.

File: PBLogging/Views/TabViews/FlightTimeTab/AuditingFlightTimeView.swift [hunk 1]
```diff
@@ -95,6 +122,31 @@ struct AuditingFlightTimeView: View {
         
     }
     
+    var popoverText: some View {
+        Text("If able, this button populates your night hours (between the end of evening civil twilight and beginning of morning civil twilight) with those calculated based on interpolated positionsand times at a 6 minute interval along a great circle line between the takeoff and landing airfields. The calculations assume a direct flight and may be inaccurate for circuitous routes.")
```
Comment (bmorris-revacomm) [inline PBLogging/Views/TabViews/FlightTimeTab/AuditingFlightTimeView.swift:126]: Need a space after "positions"

**Review by bmorris-revacomm — CHANGES_REQUESTED** [review]

File: PBLogging/Views/TabViews/FlightTimeTab/AuditingFlightTimeView.swift (hunk 1 above)
Comment (jbethanco) [reply PBLogging/Views/TabViews/FlightTimeTab/AuditingFlightTimeView.swift]: good catch.

**Review by bmorris-revacomm — APPROVED** [review]
Looks good to me 🦄

**Pull request merged**